oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/sort/ExternalSort.java

```diff
@@ -410,11 +536,19 @@ public static File sortAndSave(List<String> tmplist,
      *            location of the temporary files (set to null for default location)
      * @param distinct
      *            Pass <code>true</code> if duplicate lines should be discarded.
+     * @param filterPredicate
+     *            predicate to filter out data which need to be sorted
      */
+    public static File sortAndSave(List<String> tmplist,
+                                       Comparator<String> cmp, Charset cs, File tmpdirectory,
+                                       boolean distinct, boolean usegzip, Predicate<String> filterPredicate) throws IOException {
+        return sortAndSave(tmplist, cmp, cs, tmpdirectory, distinct, usegzip, identity(), filterPredicate);
+    }
+
     public static File sortAndSave(List<String> tmplist,
```

Comment:
Not sure why this method uses List<String> but other methods use List<T>. These methods may be already available. I didn't see any usage of these withink Oak. Maybe mark them deprecated?

Reply:
As this is from external code. So I didn't removed this. But yes there are methods we don't use.